frontend/src/pages/modelRegistry/screens/components/DeployRegisteredModelModal.tsx

```diff
@@ -153,6 +168,7 @@ const DeployRegisteredModelModal: React.FC<DeployRegisteredModelModalProps> = ({
         registeredModelDeployInfo={registeredModelDeployInfo}
         projectContext={{ currentProject: selectedProject, connections }}
         projectSection={projectSection}
+        existingUriOption={registeredModelDeployInfo.modelArtifactUri}
```

Comment:
We only want to pass `existingUriOption` in our special registry.redhat.io OCI URI case, otherwise it should be undefined. That's the only prefilling case where the user doesn't need a connection. Always passing it like this means that all models being deployed from the registry get this extra option which isn't valid in other cases, for example it exposes our nonsense S3 URIs:

![Screenshot 2025-03-06 at 6 38 17 PM](https://github.com/user-attachments/assets/928cfec5-1ced-40b5-b52b-e1cdfc879b37)

We might want an `isRedHatRegistryUri` util for our `.includes('oci://registry.redhat.io/')` so we aren't duplicating that magic string, and then we can use it here and in usePrefillDeployModalFromModelRegistry.

Reply:
Ugh, I was supposed to get to this after writing the tests 😞 . Completely slipped my mind. Thanks for pointing this out.